### PR TITLE
update to v1.12.1 and add run_exports

### LINKED ISSUE #88: Adding `run_exports` to pin `ucx` in downstream dependencies
Not sure how tightly we need to pin things. To the patch release, minor, or if major version would be safe. Raising this issue to track that discussion and update the package accordingly

### PR DESCRIPTION
**Pull request by matthiasdiener**
Fixes #88.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by leofang** [comment]
Thanks @matthiasdiener. LGTM but I am too tired today so I might not be making sense 😅 Let's wait a bit and see if anyone from @conda-forge/ucx-split can jump in...

**Comment by leofang** [comment]
@jakirkham would you have time for a look? 🙂

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,7 +10,9 @@ package:
 
 source:
   url: https://github.com/openucx/ucx/archive/v{{ ucx_version }}.tar.gz
-  sha256: e26d1696912cc4a005b93cd7fe61a8d8195e6e61bd5ded47fd232b29f6c94b8b
+  sha256: 9bef31aed0e28bf1973d28d74d9ac4f8926c43ca3b7010bd22a084e164e31b71
+  patches:
+    - 8060.patch  # Likely unnecessary after 1.12.1
```
Comment (jakirkham) [inline recipe/meta.yaml:15]: Curious, why was this patch needed?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -35,6 +37,8 @@ outputs:
     version: {{ ucx_version }}
     build:
       number: {{ number }}
+      run_exports:
+        - {{ pin_subpackage("ucx", max_pin="x.x") }}
```
Comment (jakirkham) [inline recipe/meta.yaml:41]: @pentschev, does this pinning seem reasonable to you? Basically this pins the major and minor version, but let's the patch version float

**Review by jakirkham — COMMENTED** [review]
Thanks Matthias! 😄

Had a couple questions below

File: recipe/meta.yaml (hunk 1 above)
Comment (matthiasdiener) [reply recipe/meta.yaml:15]: The patch fixes an issue in 1.12.1: https://github.com/openucx/ucx/pull/8060. A fix has been merged, so hopefully the next release won't need that patch anymore.

File: recipe/meta.yaml (hunk 2 above)
Comment (pentschev) [reply recipe/meta.yaml:41]: Sorry for the naive question, but what is this pinning exactly responsible for? If the package is building `1.12.1`, how would not including the same patch version here affect things elsewhere?

Comment (jakirkham) [reply recipe/meta.yaml:41]: It affects other packages that build against `ucx` and depend on it at runtime. So if a package like `openmpi` builds with `ucx` version `1.12.1`, they will depend on `ucx >=1.12.1,<1.13.0` at runtime.

Comment (pentschev) [reply recipe/meta.yaml:41]: Got it. I think that should be fine then, as API/ABI should be consistent among patch releases.

**Comment by leofang** [comment]
@conda-forge-admin, please rerender

**Comment by leofang** [comment]
hmmm I wonder why aarch64 and ppc64le builds are gone after rerendering...

**Comment by matthiasdiener** [comment]
> hmmm I wonder why aarch64 and ppc64le builds are gone after rerendering...

They were just migrated to Travis (instead of Azure), but they still appear to be there.

**Comment by leofang** [comment]
Ah missed that, thanks Matthias!

**Pull request merged**